src/pages/AddUnreportedExpense.tsx

```diff
@@ -106,11 +110,50 @@ function AddUnreportedExpense({route}: AddUnreportedExpensePageType) {
 
     const styles = useThemeStyles();
     const selectionListRef = useRef<SelectionListHandle>(null);
-    const sections: Array<SectionListDataType<Transaction & ListItem>> = useMemo(() => createUnreportedExpenseSections(transactions), [transactions]);
+    const filteredTransactions = useMemo(() => {
+        if (!debouncedSearchValue.trim()) {
+            return transactions;
+        }
+
+        return tokenizedSearch(transactions, debouncedSearchValue, (transaction) => {
+            const searchableFields: string[] = [];
+
+            const merchant = getMerchant(transaction);
+            if (merchant !== CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT) {
+                searchableFields.push(merchant);
+            }
+
+            const description = getDescription(transaction);
+            if (description.trim()) {
+                searchableFields.push(description);
+            }
+
+            const amount = getAmount(transaction);
+            const currency = getCurrency(transaction);
+            const formattedAmount = convertToDisplayString(amount, currency);
+            searchableFields.push(formattedAmount);
```

Comment:
We should have added a condition to normalize the amount. Not doing so caused https://github.com/Expensify/App/issues/72803